src/features/todos.ts

```diff
@@ -2,7 +1,41 @@
+import { createAsyncThunk, createSlice, PayloadAction } from '@reduxjs/toolkit';
 import { Todo } from '../types/Todo';
+import { getTodos } from '../api';
+
+type InitialState = {
+  todos: Todo[];
+  loaded: boolean;
+  selectedTodo: Todo | null;
+};
+
+const initialState: InitialState = {
+  todos: [],
+  loaded: false,
+  selectedTodo: null,
+};
+
+export const init = createAsyncThunk('todos/fetch', () => {
+  return getTodos();
+});
 
 export const todosSlice = createSlice({
   name: 'todos',
-  initialState: [] as Todo[],
-  reducers: {},
+  initialState,
+  reducers: {
+    setSelectedTodo: (state, action: PayloadAction<Todo | null>) => {
+      return { ...state, selectedTodo: action.payload };
+    },
+  },
+  extraReducers(builder) {
+    builder.addCase(init.pending, state => {
+      return { ...state, loaded: true };
+    });
+
+    builder.addCase(init.fulfilled, (state, action) => {
+      return { ...state, todos: action.payload, loaded: false };
```

Comment:
The `loaded` state should be set to `true` in the `init.fulfilled` case to indicate that data has been successfully loaded. This will ensure that the `TodoList` component is displayed once the data is available.